quaint/src/connector/mssql/native/mod.rs

```diff
@@ -59,6 +64,7 @@ pub struct Mssql {
     url: MssqlUrl,
     socket_timeout: Option<Duration>,
     is_healthy: AtomicBool,
+    transaction_depth: Arc<Mutex<i32>>,
```

Comment:
Could it be replaced it with `AtomicI32`?

Reply:
My Rust knowledge is a bit limited, but yes I think this could be an `AtomicI32`. The `transaction_depth` value is just a simple counter, so AFAIK the value should work fine.

Reply:
Additionally, is there any reason for it to be a signed `i32`? I'd prefer an `AtomicU32` here.

Reply:
@aqrln Yup, it can definitely be unsigned 👍

Reply:
@aqrln My understanding is that I can swap `Arc<Mutex<i32>>` for `Arc<AtomicU32>` as the Mutex is no longer required, and then I won't need to unwrap the value anymore. Is that correct or am I missing something?

Reply:
Yes, if you need to concurrently read and modify the same counter from multiple places, then atomics are the way to go, a mutex is a bit excessive for an integer.

By the way: I haven't read through the code thoroughly to understand the logic yet but from the cursory glance even `Arc<AtomicU32>` is a bit surprising in some places as I wouldn't expect the same exact counter to be shared in so many structs.

Reply:
@aqrln Ok good to know thanks!

> from the cursory glance even Arc<AtomicU32> is a bit surprising in some places

Could you expand on this a little bit? I'm a bit of a Rust noob 😄 
I need to add a `transaction_depth` field to every connector implementation. The depth is a simple counter that is incremented or decremented as transactions are opened and closed within the initial transaction object.

Reply:
Again, I didn't read the diff very thoroughly so I might have misunderstood, but I was a bit confused why every transaction and `TransactionOptions` references what seems to be the same transaction depth counter stored in the connector via `Arc`. How does this work with concurrent transactions on the same level of depth?

Reply:
🤔 I think my code is making the assumption that top level transaction is being run in a separate connection object, but I think that may not be correct. The transaction depth probably needs to be stored in a map of transaction_id to depth to make sure the correct depth value is used.